Add BioBlend tests

arash77 opened the pull request:
This pull request adds BioBlend tests to the project.
https://github.com/galaxyproject/bioblend/blob/main/.github/workflows/test.yaml


## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

arash77 committed:
add bioblend tests

arash77 committed:
use bioblend repo instead of put the run test file in galaxy repo

arash77 committed:
Update bioblend test script path

arash77 committed:
Format bioblend workflow

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,128 @@
+on:
+  pull_request:
+    paths:
+      - "lib/galaxy/webapps/galaxy/api/**"
```
```suggestion
      - .github/workflows/bioblend.yaml
      - lib/galaxy/schema/**
      - lib/galaxy/webapps/galaxy/api/**
      - lib/galaxy/webapps/galaxy/services/**
```

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,128 @@
+  workflow_dispatch:
+
+concurrency:
+  group: test-${{ github.ref }}
```
```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
```

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,128 @@
+  cancel-in-progress: true
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
```
```suggestion
    runs-on: ubuntu-latest
```

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,128 @@
+          fi
+          # Install Galaxy
+          GALAXY_DIR=galaxy-${{ matrix.galaxy_version }}
+          git clone --depth=1 -b ${{ matrix.galaxy_version }} https://github.com/galaxyproject/galaxy $GALAXY_DIR
```
As mentioned, get galaxy using `actions/checkout`.

```suggestion
```

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,128 @@
+          # yet: https://github.community/t/github-actions-services-available-on-others-vms/16916
+          # - os: macos-latest
+          #   tox_env: py38
+          #   galaxy_version: dev
```
You don't need this big matrix. Drop `os` and `galaxy_version` (you should use `actions/checkout` to get galaxy at the commit being tested), and choose a Python version for Galaxy (I'd say '3.8') and one for BioBlend ('3.13') to test on.

arash77 replied:
What about the `tox_env`, as it determines the BioBlend python version?

nsoranzo replied:
You can keep `tox_env: [py313]`, as the BioBlend Python version is then derived from that already in the steps below.

nsoranzo replied:
(Amended the above comments to say Python 3.13 for BioBlend).

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,128 @@
+              galaxy_python_version=3.8
+              ;;
+          esac
+          echo "galaxy_python_version=$galaxy_python_version" >> $GITHUB_OUTPUT
```
Drop this in favour of specifying `galaxy_python_version: ['3.8']` in the job matrix.

arash77 committed:
Apply suggestions
Drop galaxy_version and os from matrix and use galaxy_python_version.
Use actions/checkout instead of git clone.

Co-authored-by: Nicola Soranzo <nicola.soranzo@earlham.ac.uk>

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,81 @@
+        run: |
+          # Create a PostgreSQL database for Galaxy. The default SQLite3 database makes test fail randomly because of "database locked" error.
+          createdb -U postgres galaxy
+          export BIOBLEND_TOOLSHED_URL=https://testtoolshed.g2.bx.psu.edu/
```
No need to test BioBlend's interactions with the Test ToolShed.

```suggestion
```

arash77 committed:
No need to test BioBlend's interactions with the Test ToolShed.

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

nsoranzo commented on .github/workflows/bioblend.yaml:
```diff
@@ -0,0 +1,80 @@
+      - name: Install tox
+        run: |
+          python3 -m pip install --upgrade pip setuptools
+          python3 -m pip install 'tox>=1.8.0' 'virtualenv>=20.0.14'
```
virtualenv was only needed by old Galaxy versions.

```suggestion
          python3 -m pip install 'tox>=1.8.0'
```

nsoranzo reviewed (approved):
19 BioBlend tests fail as expected, thanks!
I'd say we merge this and make the tests pass again in a separate PR.

arash77 committed:
virtualenv was only needed by old Galaxy versions.

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>
